code/network/lib/Loot/Network/ZMQ/Common.hs

```diff
@@ -87,82 +59,84 @@ ztLog Logging{..} level t = do
 endpointTcp h p = "tcp://" <> h <> ":" <> show p
 
--- | Convenient wrapper for host + two ports. This is
--- later to be converted to 'ZTNodeId'.
-data PreZTNodeId = PreZTNodeId
-    { pztIdHost       :: !String
-    , pztIdRouterPort :: !Integer
-    , pztIdPubPort    :: !Integer
-    } deriving (Eq, Ord, Show, Generic)
-
--- | Parser of 'ZTNodeId' in form of "host:port1:port2".
-parsePreZTNodeId :: String -> Either String PreZTNodeId
-parsePreZTNodeId s = case splitBy ':' s of
-    [ztHost,p1,p2] ->
-        case (readMaybe p1, readMaybe p2) of
-            (Just port1, Just port2) -> Right $ PreZTNodeId ztHost port1 port2
-            _                        -> Left "Can't parse either of the ports"
-    _            -> Left "String should have expactly two columns"
-  where
-    splitBy :: Eq a => a -> [a] -> [[a]]
-    splitBy _ [] = []
-    splitBy d s' = x : splitBy d (drop 1 y) where (x,y) = L.span (/= d) s'
+----------------------------------------------------------------------------
+-- Node identifiers
+----------------------------------------------------------------------------
 
 -- | NodeId as seen in ZMQ TCP.
 data ZTNodeId = ZTNodeId
     { ztIdHost       :: !String  -- ^ Host, can be domain name or ip address.
                                  -- This is what user specifies and it's used for
                                  -- logging only. Internal code uses internal id.
-    , ztIdInternal   :: !String  -- ^ IP address, must match the resolved host.
-                                 -- It is used as a node identifier.
     , ztIdRouterPort :: !Integer -- ^ Port for ROUTER socket.
     , ztIdPubPort    :: !Integer -- ^ Port for PUB socket.
     } deriving (Eq, Ord, Show, Generic)
 
 instance Serialise ZTNodeId
+instance Hashable ZTNodeId
 
--- | Creates a proper 'ZTNodeId' from 'PreZTNodeId'. The host inside should not
--- be a wildcard, but a ip address or a domain name only.
-mkZTNodeId :: PreZTNodeId -> IO ZTNodeId
-mkZTNodeId (PreZTNodeId ztIdHost ztIdRouterPort ztIdPubPort) = do
-    ztIdInternal <- resolveHost ztIdHost
-    pure $ ZTNodeId {..}
+-- | Parser of 'ZTNodeId' in form of "host:port1:port2".
+parseZTNodeId :: String -> Either String ZTNodeId
+parseZTNodeId s = case splitBy ':' s of
+    [ztHost,p1,p2] ->
+        case (readMaybe p1, readMaybe p2) of
+            (Just port1, Just port2) -> Right $ ZTNodeId ztHost port1 port2
+            _                        -> Left "Can't parse either of the ports"
+    _            -> Left "String should have expactly two columns"
   where
-    -- Resolves given host. This is needed to unify nodes' identifiers -- all
-    -- socket ids are IPv4 addresses, not hosts.
-    resolveHost :: String -> IO String
-    resolveHost address = do
-        ent <- getHostByName address
-        inet_ntoa (hostAddress ent)
+    splitBy :: Eq a => a -> [a] -> [[a]]
+    splitBy _ [] = []
+    splitBy d s' = x : splitBy d (drop 1 y) where (x,y) = L.span (/= d) s'
```

Comment:
[splitOn](http://hackage.haskell.org/package/split-0.2.3.3/docs/Data-List-Split.html#v:splitOn)

Reply:
Nice function, though it in some external library so I obviously won't import a lib for just a oneliner.